### PR TITLE
Added additional debug logging

### PR DESCRIPTION
**Pull request by galexrt**
This adds additional debug output at some points in the code.

### CODE REVIEW

File: pkg/ingress/tls.go [hunk 1]
```diff
@@ -128,5 +129,11 @@ func (i *Tls) RequestCert() error {
 
 	s.Object().Data = certData
 
-	return s.Save()
+	fmt.Printf("secret object data: %v\n", s)
```
Comment (gianrubio) [inline pkg/ingress/tls.go]: change ```fmt.printf``` to ```...Log().Debugf```

File: pkg/ingress/tls.go [hunk 2]
```diff
@@ -128,5 +129,11 @@ func (i *Tls) RequestCert() error {
...
+
+	returnVal := s.Save()
+
+	fmt.Printf("secret save error: %v\n", returnVal)
```
Comment (gianrubio) [inline pkg/ingress/tls.go]: Please test the returnVal before printing, also use ````Log().Debugf```

File: pkg/secret/secret.go [hunk 3]
```diff
@@ -65,7 +65,8 @@ func (o *Secret) Save() (err error) {
 		obj, err = o.client().Create(o.SecretApi)
 	}
 	if err != nil {
-		return
+		o.Log().Fatalf("error creating/updating secret: %v", err.Error())
```
Comment (gianrubio) [inline pkg/secret/secret.go]: Fatalf it's not a good point here, maybe the api are unreachable during this time why crash kube-lego?

File: pkg/secret/secret.go [hunk 4]
```diff
@@ -65,7 +65,8 @@ func (o *Secret) Save() (err error) {
...
+		return err
```
Comment (gianrubio) [inline pkg/secret/secret.go]: Or return the error or print the log

File: pkg/ingress/tls.go (hunk 2 above)
Comment (galexrt) [inline pkg/ingress/tls.go]: @gianrubio How should I test it? The `..Debugf()` functions `%v` doesn't have a problem with nil value and others.

**Comment by galexrt** [comment]
Closing as this PR had been opened a long time ago.

**Pull request closed without merging**